### ISSUE TITLE
Typo tolerance disableOnAttributes ineffective for attributes with similar content

### ISSUE THREAD

**Issue by Florian86** [meilisearch/meilisearch#4256]
Hey, I'm having trouble configuring/understanding the typo tolerance setting `disableOnAttributes`. Not sure if this is a bug or just behavior that I don't understand.

I added the following documents:

```json
[
  {
    "id": "1",
    "attr1": "Meilisearch",
    "attr2": "hello Meilisearch"
  },
  {
    "id": "2",
    "attr1": "Meilisearch",
    "attr2": "foobar"
  }
]
```

With default configuration I see the following result when searching for `Meilisearch`:

```json
{
  "hits": [
    {
      "id": "1",
      "attr1": "Meilisearch",
      "attr2": "hello Meilisearch",
    },
    {
      "id": "2",
      "attr1": "Meilisearch",
      "attr2": "foobar"
    }
  ],
  "query": "Meilisearch",
}
```

Searching for `Meilisearcx` results in:

```json
{
  "hits": [
    {
      "attr1": "Meilisearch",
      "attr2": "hello Meilisearch",
      "id": "1"
    },
    {
      "attr1": "Meilisearch",
      "attr2": "foobar",
      "id": "2"
    }
  ],
  "query": "Meilisearcx",
}
```

Everything as expected so far.

When limiting the searchable attributes to `attr1` and disabling typo tolerance on `attr1`, the results change as expected.
Searching for `Meilisearch` => 2 hits.
Searching for `Meilisearcx` => 0 hits.

Now I add `attr2` to searchable attributes and disable typo tolerance on `attr2` as well.

```json
{
  "enabled": true,
  "minWordSizeForTypos": {
    "oneTypo": 5,
    "twoTypos": 9
  },
  "disableOnWords": [],
  "disableOnAttributes": [
    "attr1",
    "attr2"
  ]
}
```

Searching for `Meilisearch` => 2 hits, as expected.

But when searching for `Meilisearcx`, I would have expected to get no results. Instead, I get one hit, although typo tolerance is disabled on `attr1` and `attr2`:

```json
{
  "hits": [
    {
      "attr1": "Meilisearch",
      "attr2": "hello Meilisearch",
      "id": "1"
    }
  ],
  "query": "Meilisearcx",
}
```

When changing the documents to:

```json
[
  {
    "id": "1",
    "attr1": "Meilisearch",
    "attr2": "something else"
  },
  {
    "id": "2",
    "attr1": "Meilisearch",
    "attr2": "foobar"
  }
]
```

I do not get any results, as I would expect.

So in short: Disabling typo tolerance on two attributes that have similar content does not seem to work.

**Meilisearch version:** v1.5.1

**Comment by ManyTheFish** [comment]
Hello @Florian86,
Indeed, it's a bug,
I reproduced the bug on v1.5.0 and v1.5.1.
However, the bug doesn't seem to appear on the main branch (future v1.6.0).
So this bug should be fixed in the next release (v1.6.0).

The first release candidate has already been released; don't hesitate to try it:
https://github.com/meilisearch/meilisearch/releases

If you manage to reproduce the bug with the release candidate,
please reopen the issue, and we'll make further investigation,

Thank you for the report!
see you!

> Closing this issue as fixed in v1.6.0
